chore: address commit comments (#5894)

JaySoni1 opened the pull request:
## Description  
This PR addresses the feedback provided on commit [`c31eccd`](https://github.com/stdlib-js/stdlib/commit/c31eccd4e752a5647b718df9f3b70739c8a4b45b), specifically the review comment regarding the JSDoc typedef for `ComplexArray`.

## Changes Made  
- Reverted the incorrect modification to the JSDoc type annotation on **Line 103**.  
- Restored `ComplexArray` as a defined JSDoc type, per the suggestion.  
- Confirmed that the linting error was a false positive and no further change was required.

## Files Updated  
- `stdlib/lib/node_modules/@stdlib/array/filled/lib/main.js`

## Context  
The review comment referenced:  
> **Line 103**: This change should be reverted. `ComplexArray` is a defined JSDoc type.  
> See `stdlib/tools/docs/jsdoc/typedefs/arrays.js`.

✅ Verified that `ComplexArray` is properly defined in `typedefs/arrays.js`, and the JSDoc usage is now consistent with project standards.

## Related Issue  
Closes #5894  

## Checklist  
- [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)  
- [x] Followed the contributing guidelines  
- [x] Made changes in a separate branch from `develop`  
- [x] Passed all existing tests  
- [x] Verified linting and formatting compliance

JaySoni1 committed:
Reverted ComplexArray

Signed-off-by: JaySoni1 <jaygsoni14@gmail.com>

JaySoni1 commented:
@Planeshifter , @kgryte please review

kgryte commented on lib/node_modules/@stdlib/array/filled/lib/main.js:
```diff
@@ -100,7 +100,7 @@ function filledAccessors( arr, value ) {
 * @throws {TypeError} must provide a length, typed array, array-like object, buffer, or iterable
 * @throws {Error} creating a generic array from an `ArrayBuffer` is not supported
-* @returns {(TypedArray|Array|Complex64Array)} array or typed array
+* @returns {(TypedArray|Array|ComplexArray|ComplexTypedArray)} array or typed array
```
```suggestion
* @returns {(TypedArray|Array|ComplexArray)} array or typed array
```

kgryte committed:
docs: remove extra type

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
style: disable lint rule

This commit disables a lint rule due to a false positive arising from `ComplexArray`. This is a defined type currently in our JSDoc type definitions.

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
style: re-enable lint rule

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into Issue5894
